src/server/index.js

```diff
@@ -30,18 +31,19 @@ async function NextAuth (req, res, userSuppliedOptions) {
     // This is passed to all methods that handle responses, and must be called
     // when they are complete so that the serverless function knows when it is
     // safe to return and that no more data will be sent.
-    const done = resolve
+    // REVIEW: Why not just call res.end() as is, and remove the Promise wrapper?
+    res.end = () => {
+      resolve()
+      res.end()
+    }
```

Comment:
Don't really understand the need for Promise and `resolve()` here. `res.end()` should already signal that the response is complete.

See: https://nodejs.org/api/http.html#http_response_end_data_encoding_callback